contracts/buyAndBurn/BuyAndBurnSwapImplementation.sol

```diff
@@ -0,0 +1,402 @@
+
+            // We only add the leftover tokenIn to LP (greedy strategy prioritizes burn over LP)
+            // Uniswap V3 can handle one-sided liquidity when price is outside range
+            uint256 amount0Desired = zeroForOne ? amountLpIn : 0;
+            uint256 amount1Desired = zeroForOne ? 0 : amountLpIn;
+
+            // Check balances
+            uint256 token0Balance = IERC20(token0).balanceOf(address(this));
+            uint256 token1Balance = IERC20(token1).balanceOf(address(this));
+
+            // Verify we have sufficient balance for what we're trying to add
+            require(
+                token0Balance >= amount0Desired,
+                BuyAndBurnSwap__InsufficientAmount(token0, amount0Desired, token0Balance)
+            );
+            require(
+                token1Balance >= amount1Desired,
+                BuyAndBurnSwap__InsufficientAmount(token1, amount1Desired, token1Balance)
+            );
+
+            // Approve position manager to spend tokens
+            if (amount0Desired > 0) {
+                IERC20(token0).forceApprove(address(positionManager), amount0Desired);
+            }
+            if (amount1Desired > 0) {
+                IERC20(token1).forceApprove(address(positionManager), amount1Desired);
+            }
+
+            // Try to add liquidity with leftover tokenIn
+            // This may fail if amount is too small or position requires both tokens
+            // In that case, we'll just treat the leftover as spare (aligned with greedy strategy)
+            try positionManager.increaseLiquidity(
+                INonfungiblePositionManager.IncreaseLiquidityParams({
+                    tokenId: params.lpTokenId,
+                    amount0Desired: amount0Desired,
+                    amount1Desired: amount1Desired,
+                    amount0Min: 0, // No minimum to allow price movement
+                    amount1Min: 0, // No minimum to allow price movement
+                    deadline: block.timestamp
+                })
+            ) returns (uint128 liquidity, uint256 amount0, uint256 amount1) {
+                // Successfully added liquidity
+                liquidityDelta = liquidity;
+
+                // Calculate spare tokenIn (what wasn't used for LP)
+                spareIn = zeroForOne ? amount0Desired - amount0 : amount1Desired - amount1;
+            } catch {
+                // Failed to add liquidity (amount too small, position constraints, etc.)
+                // Treat all leftover tokenIn as spare - this aligns with greedy strategy
+                // where LP is optional and burn is the priority
+                spareIn = amountLpIn;
+                liquidityDelta = 0;
+            }
+        } else {
+            // No leftover tokenIn, so nothing to add to LP
+            spareIn = 0;
+            liquidityDelta = 0;
+        }
+
+        // ALL tokenOut goes to burn (not used for LP in greedy strategy)
+        spareOut = amountSwapOut;
```

Comment:
There is no `return` in `if (amountLpIn > 0) { ... }`. If there is LP added, this calculation of `spareOut` is incorrect.